currency-rate/internal/broker/rate/client.go

```diff
@@ -0,0 +1,95 @@
+package rate
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"strconv"
+	"time"
+
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/broker"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/broker/transport"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/broker/transport/config"
+	"github.com/VictoriaMetrics/metrics"
+)
+
+var lastEventID int
+
+var logger *slog.Logger
+
+var (
+	totalSentRateMessagesCounter = metrics.NewCounter(
+		`broker_sent_messages_total{producer="rate_producer"}`,
+	)
+	messageSizeHistogram = metrics.NewHistogram(
+		`broker_message_size_bytes{producer="rate_producer"}`,
+	)
+)
+
+func getLogger() *slog.Logger {
+	if logger == nil {
+		logger = slog.Default().With(slog.Any("src", "rateProducer"))
```

Comment:
what is this function doing? Kind of singleton? I'm not really fun of `init` function in GO, but maybe it will suite better here?